src/logic/queryParams.js

```diff
@@ -35,7 +36,25 @@ import {
  * @return {string}
  */
 const getSearchQueryFromQueryParameterOrLocalStorage = route => {
-  const { [CommonQueryParameters.SearchFilters]: sq } = route?.query;
+  const {
+    [CommonQueryParameters.SearchFilters]: sq,
+    f,
```

Comment:
I was just going to comment that it's great that you are using global constants for query parameters and now this code is reverted back to hardcoded `f` 😢

Reply:
but but... you brought the opposite argument just a few comments ago :D 
`f` is something that we don't support and it is not repeated in the code, so why should we use a global constant for that?
> Having it here sends a wrong message that we support both query parameters even on the pages that never supported f parameter before, when in fact this is just a backward compatibility with old versions

Reply:
That's not what I meant. My suggestion was to introduce a separate function to extract search query from `f` parameter _only_ where it was used before (search and search image pages). This would make it clear: it's a legacy parameter support. Right now this function looks confusing without explicit comments: 
 * one parameter is used with a constant and the other one is hardcoded without explanation. 
 * one parameter has higher priority but it is not explained why it is here.

Filters management is such a fundamental part of the codebase and we have all the opportunities to make it easy to understand and bug free before it is introduced in production. We should use them.

Reply:
Ah ok, I understand. Unfortunately I cannot work on this tomorrow, so I have to leave the final touch to you :D

Reply:
BTW Is it really necessary to bring this code to the `searchPage` and to the `searchImage` page? I still think its place is here in the getter function (and it takes less than 10 lines). The code on those components will be less readable.

Reply:
This is a suggestion. If you think the code needs to stay here and be available on any page that uses this code then leave it here. But I'd still recommend to use a global query parameter name (something like `CommonQueryParameters.LegacySearchQuery`) to keep all of global parameters in one place for reference.

Reply:
thanks @theorm  this solution is a good compromise!